src/components/application_manager/src/hmi_language_handler.cc

```diff
@@ -288,11 +290,14 @@ void HMILanguageHandler::HandleWrongLanguageApp(const Apps::value_type& app) {
                       << " is not found within apps with wrong language.");
     return;
   }
+  apps_.erase(it);
+  if (0 == apps_.size()) {
+    LOG4CXX_DEBUG(logger_,
+                  "All apps processed. Unsubscribing from all events.");
+    unsubscribe_from_all_events();
```

Comment:
@dtrunov 
Consider case when application registered, but SDL has not get all languages(VR,TTS, UI) from HMI yet.